www/src/content/blog/2024-11-28-client-side-guide.mdx

```diff
@@ -0,0 +1,613 @@
+---
+title: "Step-by-step guide: Adding client-side logic to your Hono app"
+description: A guide to adding client-side logic to your Hono app.
+slug: client-side-guide
+date: 2024-11-28
+author: Oscar van Zijverden
+tags:
+  - guide
+  - Hono
+  - hono.js
+  - typescript
+  - hydration
+---
+
+import {
+  Aside,
+  Card,
+  FileTree,
+  TabItem,
+  Tabs
+} from "@astrojs/starlight/components";
+import videoHydrated from "../../assets/blog/2024-11-28-client-side-guide-hydrated.mov";
+import videoStatic from "../../assets/blog/2024-11-28-client-side-guide-static.mov";
+
+[Hono](https://hono.dev) is a great framework to build serverless apps with
+familiar APIs using Web Standards. It comes with a ton of features out of the
+box.
+
+
+
+One of these features is the ability to compose & render HTML server-side, which
+is great for static content.\
+But what if you want to add some client-side logic to your app?
+You can do so by using the hooks provided by Hono! However, they might not work
+in your browser — that's because we're not shipping any JavaScript to the
+browser!
+
+In this guide we'll go over how to build an Hono app with client-side logic,
+unlocking the full potential of your projects.
+
+## What are we building?
+
+We're building a simple app that renders a counter component server-side and
+hydrates it client-side.\
```

Comment:
might be good to explain hydration for the uninitiated. (a parenthetical would suffice — just mentioning that hydration refers to filling it up with javascript)